spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc

```diff
@@ -0,0 +1,99 @@
+=== Channel Adapters for Google Cloud Storage
+
+The channel adapters for Google Cloud Storage allow you to read and write files to Google Cloud
+Storage through `MessageChannels`.
+
+Spring Cloud GCP provides two inbound adapters, `GcsInboundFileSynchronizingMessageSource` and
+`GcsStreamingMessageSource`, and one outbound adapter, `GcsMessageHandler`.
+
+To use the Storage portion of Spring Integration for Spring Cloud GCP, you must also provide the
+`spring-integration-file` dependency, since it isn't going to be pulled transitively.
+
+Maven coordinates, using Spring Cloud GCP BOM:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.cloud</groupId>
+    <artifactId>spring-cloud-gcp-integration</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.integration</groupId>
+    <artifactId>spring-integration-file</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source,subs="normal"]
+----
+dependencies {
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-integration', version: '{project-version}'
+    compile group: 'org.springframework.integration', name: 'spring-integration-file', version: '{spring-integration-version}'
```

Comment:
Well, I think we don't need version here at all - Spring Integration is managed by Spring Boot. Only what you need is to add appropriate `spring-boot` plugin to your Gradle config. 
Anyway I would leave all of these tricks and hooks up to end-user and appropriate project references like Spring Boot.
That is not our goal here to cover all the Spring portfolio in one doc.

Reply:
Indeed, which is why I prefer to add the version straight up.

One thing is that while we don't upgrade to SI 5 on our side, we should probably recommend users to use 4.3.13.
I might settle for 4.3.13.

Reply:
??? That's not good. You eliminate the purpose of Spring Boot at all.

And again: that is the part which you would have to support in the future - fix the version from upgrade to upgrade.

Reply:
I thought you were suggesting against advising for tricks like using the gradle bom plugin here.
Which only leaves us with the option of leaving the version, if I understand correctly?

If there is a way to get rid of the version here other than adding gradle plugins, I would like to pursue it!

Reply:
😕 . I said:
> we don't need version here at all

AFAIK there is no other way for Gradle. However you still stick with something for Maven as well. 
And don't forget that there are some other building and dependency management tools and you don't cover them here. I would stick away not only from versions (see my comment about future support), but also from the dependency management tool. Any way it is not our project responsibility to demonstrate how to manage deps and their versions. Too much noise and no concentration on the purpose of the feature we are describing. As a middle ground we may send people to Spring Boot docs on the matter - in most cases people these days start from there for their projects.

Reply:
The goal has always been for users to be able to copy and paste snippets from the refdoc to their apps. That wouldn't work with Gradle if we were to omit versions.

The reason the Maven snippets don't contain versions and Gradle ones do is because [we say we will omit them](https://docs.spring.io/spring-cloud-gcp/docs/1.0.0.M1/reference/htmlsingle/#_dependency_management) earlier on.

BOM isn't a Gradle concept, so I chose to show the versions for those.

We can change that and hide versions for Gradle as well if we want to, but I'd prefer to do that in another PR.

Reply:
As a matter of fact, now that I'm having to also manually add the `google-cloud-storage` version, I think I'll want to say in the dependency management chapter that we will hide versions for Gradle dependencies too.
I'll open an issue for that.

Reply:
Well, since you show there how to use BOM in Maven that's not a big deal to show the same for Gradle I guess, since you already mention `dependency-management-plugin`. 
And I don't think that RM is a copy/paste guide. You definitely still miss a lot of pieces in these samples and end-user still has to do something, for example you don't explain where to place all those dependencies etc. etc. We have samples applications in the end! Why just don't point there rather than pursue this pain how to resolve current SI version for our Gradle snippet in the manual?

That's my last comment on the matter and my general idea here is to avoid routine support as much as possible. 

And that's somehow too much confronting points of view how project should be developed...

Reply:
Oh! Sorry, forgot to mention. If you still insist, let it be: I really spent already too much time to convince you in different vision. Gone to do some other useful work...

Sorry for the noise

Reply:
@artembilan you convinced me!
I hated having to keep those versions around and manually maintaining in the first place. I'll take care of it after this PR, as I said.